From 110cde3cfaafc3b8704aeb0bf7c6e3c46eef7501 Mon Sep 17 00:00:00 2001
From: Tim Harvey <tharvey@gateworks.com>
Date: Tue, 15 Sep 2015 10:53:04 -0700
Subject: [PATCH] input: misc: Add Gateworks System Controller input support

The Gateworks System controller can interrupt the host processor on the
following events:
 * User Pushbutton press/release/held
 * Secure Key erase
 * EEPROM Write-Protect error
 * GPIO Change event
 * Tamper detect

This patch adds a Linux input driver that will expose these events as Linux
Input Events.

See http://trac.gateworks.com/wiki/gsc

Signed-off-by: Tim Harvey <tharvey@gateworks.com>
---
 drivers/input/misc/Kconfig     |  11 +++
 drivers/input/misc/Makefile    |   1 +
 drivers/input/misc/gsc-input.c | 180 +++++++++++++++++++++++++++++++++++++++++
 3 files changed, 192 insertions(+)
 create mode 100644 drivers/input/misc/gsc-input.c

--- a/drivers/input/misc/Kconfig
+++ b/drivers/input/misc/Kconfig
@@ -103,6 +103,17 @@ config INPUT_E3X0_BUTTON
 	  To compile this driver as a module, choose M here: the
 	  module will be called e3x0_button.
 
+config INPUT_GSC
+	tristate "Gateworks System Controller Input support"
+	depends on GSC_CORE
+	help
+	Support for Gateworks System Controller input device.
+	  Say Y here if you want to enable Linux input device support for
+	  GSC user-pushbutton, and tamper events.
+
+	  To compile this driver as a module, choose M here. The module will
+	  be called gsc_input.
+
 config INPUT_PCSPKR
 	tristate "PC Speaker support"
 	depends on PCSPKR_PLATFORM
--- a/drivers/input/misc/Makefile
+++ b/drivers/input/misc/Makefile
@@ -50,6 +50,7 @@ obj-$(CONFIG_INPUT_PALMAS_PWRBUTTON)	+=
 obj-$(CONFIG_INPUT_PCAP)		+= pcap_keys.o
 obj-$(CONFIG_INPUT_PCF50633_PMU)	+= pcf50633-input.o
 obj-$(CONFIG_INPUT_PCF8574)		+= pcf8574_keypad.o
+obj-$(CONFIG_INPUT_GSC)			+= gsc-input.o
 obj-$(CONFIG_INPUT_PCSPKR)		+= pcspkr.o
 obj-$(CONFIG_INPUT_PM8941_PWRKEY)	+= pm8941-pwrkey.o
 obj-$(CONFIG_INPUT_PM8XXX_VIBRATOR)	+= pm8xxx-vibrator.o
